src/foam/lib/PropertyPredicate.js

```diff
@@ -0,0 +1,26 @@
+foam.INTERFACE({
+  package: 'foam.lib',
+  name: 'PropertyPredicate',
+
+  methods: [
+    {
+      name: 'propertyPredicateCheck',
+      type: 'boolean',
```

Comment:
type: 'Boolean'

The bulk of the type resolution happens in here: https://github.com/foam-framework/foam2/blob/master/src/foam/core/Type.js

Primitives like Boolean, Integer, Float, etc. are special and know how to output in both swift and java. Use those names when referring to primitives. Use either those special types or model IDs only. If it's not a modelled object, use javaType (assuming you're doing java dev).